tests/e2e/lib.sh

```diff
@@ -454,9 +454,13 @@ check_for_errors_in_stackrox_logs() {
     # shellcheck disable=SC2010,SC2086
     filtered=$(ls $logs | grep -Ev "(previous|_describe).log$" || true)
     if [[ -n "$filtered" ]]; then
+        local check_out=""
         # shellcheck disable=SC2086
-        if ! scripts/ci/logcheck/check.sh $filtered; then
+        if ! check_out="$(scripts/ci/logcheck/check.sh $filtered)"; then
+            save_junit_failure "SuspiciousLog" "Suspicious entries in log file(s)" "$check_out"
```

Comment:
Can I ask you to also change this fragment to wrap `${details}` in XML CDATA?
https://github.com/stackrox/stackrox/blob/883967db3a31301a58970af4a909ff8a85499850/scripts/ci/lib.sh#L1401-L1405

This would be:
```xml
<testsuite name="${class}" tests="1" skipped="0" failures="1" errors="0">
    <testcase name="${description}" classname="${class}">
        <failure><![CDATA[${details}]]></failure>
    </testcase>
</testsuite>
```

The reason is that CDATA allows having `<`, `>` and other characters that are otherwise considered as special in XML and which presence may mess up parsing.

---

Additionally, WDYT about including a timestamp in the file name in this line?
https://github.com/stackrox/stackrox/blob/883967db3a31301a58970af4a909ff8a85499850/scripts/ci/lib.sh#L1400

Same about `save_junit_success` function.
This will help us not be afraid that reusing the "class name" will lead to overwriting the file created by some other step.

Reply:
It is a bit outside the scope of this PR but I see the value :( I will need to test that having >1 junits for the same class is supported well by prow/spyglass.